layer.py

```diff
@@ -171,16 +192,16 @@ def cloud_action(self):
     def cloud_action(self, action):
         self._cloud_action = action
 
-    def photo_naming(self, field_name: str) -> str:
-        return self._photo_naming.get(
+    def attachment_naming(self, field_name, attachment_type: int) -> str:
+        return self._attachment_naming.get(
             field_name,
-            "'DCIM/{layername}_' || format_date(now(),'yyyyMMddhhmmsszzz') || '.jpg'".format(
+            self.ATTACHMENT_EXPRESSIONS[self.AttachmentType(attachment_type)].format(
```

Comment:
```suggestion
            self.ATTACHMENT_EXPRESSIONS[attachment_type].format(
```

So this can be super straightforward.